ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java

```diff
@@ -458,6 +459,29 @@ public void testReservedRowKeyAlias() {
     KsqlParserTestUtil.buildSingleAst("SELECT C2 as ROWKEY FROM test1 t1;", metaStore);
   }
 
+  @Test
+  public void shouldThrowOnNonAlphanumericSourceName() {
+    // Expect:
+    expectedException.expect(ParseFailedException.class);
+    expectedException.expectMessage("Got: 'foo!bar'");
+
+    // When:
+    KsqlParserTestUtil.buildSingleAst(
+        "CREATE STREAM `foo!bar` WITH (kafka_topic='foo', value_format='AVRO');",
+        metaStore);
+  }
+
+  @Test
+  public void shouldAllowEscapedTerminateQuery() {
```

Comment:
Should we also add a test for `INSERT INTO` with a quoted source name, either here or as a QTT?